service/topology_coordinator.cc

```diff
@@ -1176,8 +1187,14 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
     }
 
     future<> generate_migration_updates(std::vector<canonical_mutation>& out, const group0_guard& guard, const migration_plan& plan) {
+        const auto& tables_waiting_for_resize_finalization = plan.resize_plan().finalize_resize;
         for (const tablet_migration_info& mig : plan.migrations()) {
             co_await coroutine::maybe_yield();
+            if (tables_waiting_for_resize_finalization.contains(mig.tablet.table) && !plan.has_nodes_to_drain()) {
```

Comment:
@tgrabiec If we have pending resize finalization, perhaps we shouldn't schedule any new migrations for any table, unless we're draining. Since migration of any table blocks resize finalization. If you want to finalize resize at table B, but still allowing new migrations in table A, resize finalization finalization is blocked until you're done with all migrations in other tables. That's not really what we want, I think.

Reply:
Yes. @Lakshmipathi, that needs to be fixed. I'd rather make migrations in non-finalizing tables don't block finalization of tables which don't have migrations, but we can start with a broader filter, just make sure the filtering here is consistent with the condition for scheduling finalization.

Reply:
@raphaelsc @tgrabiec I have updated the PR. The new version doesn't schedule migrations in any table if there are resize finalizations pending. With this, we do not need to handle resize finalization within tablet migration handling. So I have removed those patches from this PR entirely. If required, I could prepare and submit them as a separate PR. Let me know if this is okay.